### PR TITLE
docker: Add Dockerstatic File for Debian 11.

### PR DESCRIPTION
**Pull request by steelhead31**
##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [X] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [X] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

**Comment by steelhead31** [comment]
JDK copied to docker container, and quick aqa test run against JDK8  
  
TEST TARGETS SUMMARY  
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++  
PASSED test targets:  
MBCS\_Tests\_pref\_ja\_JP\_linux\_0

TOTAL: 1   EXECUTED: 1   PASSED: 1   FAILED: 0   DISABLED: 0   SKIPPED: 0  
ALL TESTS PASSED  
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

\_MBCS\_Tests\_pref\_ja\_JP\_linux\_0 done

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM debian:bullseye
+# Install Base Requirements
+RUN apt-get update && apt-get install -y perl openssh-server unzip wget
+RUN echo "y" | ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key -P ""
+RUN apt update
+# Install java8 via WGET
+RUN wget -q 'https://api.adoptopenjdk.net/v2/binary/releases/openjdk8?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&type=jdk' -O /tmp/jdk8.tar.gz
```
[the remark this answers is not in the document]
Comment (karianna) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11:7]: Would be good to fix others if this was a C&P

**Review by karianna — CHANGES_REQUESTED** [review]

**Review by Haroon-Khel — COMMENTED** [review]
Aside from Martijn's changes, it looks good. I'll get a deb11 container running off these changes and then approve

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11 (hunk 1 above)
[the remark this answers is not in the document]
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11]: Fixed this, was a C&P error on my part, the other dockerstatic files have been updated in a previous PR.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11 [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+RUN mkdir -p /usr/lib/jvm/jdk8 && tar -xpzf /tmp/jdk8.tar.gz -C /usr/lib/jvm/jdk8 --strip-components=1
+# Install ant via WGET
+RUN wget -q -O /tmp/ant.zip 'https://archive.apache.org/dist/ant/binaries/apache-ant-1.10.5-bin.zip'
```
[the remark this answers is not in the document]
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11]: Ant & Ant Contrib updated to latest versions.

**Review by karianna — APPROVED** [review]
LGTM

**Comment by steelhead31** [comment]
> Aside from Martijn's changes, it looks good. I'll get a deb11 container running off these changes and then approve

@Haroon-Khel all good now, thanks!

**Review by Haroon-Khel — APPROVED** [review]
Got the container up and running at https://ci.adoptopenjdk.net/computer/test-docker-debian11-x64-1/

**Comment by Haroon-Khel** [comment]
@steelhead31 
While running a grinder on the container, at https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5048/console, it hit this error
```
17:21:54  Buildfile: /home/jenkins/workspace/Grinder/aqa-tests/TKG/scripts/build_test.xml
17:21:55    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
17:21:55    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
```
You think it could be related to the updated versions of Ant & Ant Contrib?

**Review by Haroon-Khel — CHANGES_REQUESTED** [review]
@steelhead31 
While running a grinder on the container, at https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5048/console, it hit this error
```
17:21:54  Buildfile: /home/jenkins/workspace/Grinder/aqa-tests/TKG/scripts/build_test.xml
17:21:55    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
17:21:55    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
```
You think it could be related to the updated versions of Ant & Ant Contrib?

**Comment by steelhead31** [comment]
> @steelhead31 While running a grinder on the container, at https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5048/console, it hit this error
> 
> ```
> 17:21:54  Buildfile: /home/jenkins/workspace/Grinder/aqa-tests/TKG/scripts/build_test.xml
> 17:21:55    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
> 17:21:55    [taskdef] Could not load definitions from resource net/sf/antcontrib/antlib.xml. It could not be found.
> ```
> 
> You think it could be related to the updated versions of Ant & Ant Contrib?

Yup, almost certainly is.. I'll do some investigation and try downgrading ant-contrib as I think thats likely the source of the problem.

**Comment by steelhead31** [comment]
@Haroon-Khel would you mind re-spinning the docker and running another test, it seems much happier with the old version of ant-contrib. I'll log a separate issue to investigate what will be required to upgrade to the new ant-contrib.

**Comment by Haroon-Khel** [comment]
New grinder running on the rebuilt image/container https://ci.adoptopenjdk.net/job/Grinder/5055/console

**Review by Haroon-Khel — APPROVED** [review]
Grinder passed. LGTM

**Pull request merged**